### ISSUE TITLE
Add lock cursor hotkey

### ISSUE THREAD

**Issue by mak448a** [waydroid/waydroid#957]
**Is your feature request related to a problem? Please describe.**
When playing any FPS game on Roblox, (i.e. Frontlines) the cursor doesn't get locked.

**Describe the solution you'd like**
Add a hotkey to lock the cursor to an app as described in [a comment on #482](https://github.com/waydroid/waydroid/issues/482#issuecomment-1556171257)

**Comment by ios7jbpro** [comment]
i agree with this, also happens in here on roblox, as wine is now no longer usable for it

**Comment by ios7jbpro** [comment]
i meant add some sort of switch so the mouse doesnt SLIDE OUT of the game
as roblox doesnt lock it

**Comment by ios7jbpro** [comment]
> This problem has got to do with this topic: https://devforum.roblox.com/t/android-pointer-capture-support/597829/3
> 
> The truth is, since Roblox doesn't expect a cursor on a mobile device, it has no support for capturing that cursor, which is why the pointer is free when you move the camera around.
> 
>  broken-cursor.webm 
> **THE CURSOR STAYS LOCKED WHEN RIGHT-CLICK IS PRESSED, BUT IT GOES BACK INTO PLACE WHEN RIGHT-CLICK IS RELEASED**
> 
> While it appears the cursor is locked, it is actually freely moving around normally, though we don't get to see it.
> 
> This problem won't be easily solved. It will be required for an update on Roblox's part to happen.

also update about this:
windows microsoft store edition has same issue while there is already a fix for it. so yeah this is ABSOLUTELY fixable.
https://github.com/mengdeveloper/microsoftrbx-cursor-fix

**Comment by no-venv** [comment]
@ios7jbpro , @PhoenixCausesOof , @mak448a 

created something similar to https://github.com/mengdeveloper/microsoftrbx-cursor-fix, but for linux

https://github.com/no-venv/LockRblxCursor_Linux/tree/main

**Comment by ios7jbpro** [comment]
@no-venv i love you
and about weston and x client only, considering its based on xdotool, id suggest to give ydotool a try, its for wayland, it doesnt work as well as xdotool though afaik

**Comment by no-venv** [comment]
> @no-venv i love you and about weston and x client only, considering its based on xdotool, id suggest to give ydotool a try, its for wayland, it doesnt work as well as xdotool though afaik

`pynput` is actually used for handing keyboard and mouse events, it works under wayland.

however, the main problem is on getting the window size + position (so the cursor wraps around) on wayland. 

it seems like there isn't any specific way that is universal, i saw one for gnome but not for kde and the entire process is just soo complicated

**Comment by BrandowLucas** [comment]
+1

**Comment by Robbidy** [comment]
Don't really know if this is the best place to comment, but I was wondering how you would implement the cursor hotkey into waydroid. I don't really know much about python so I don't know how you run that code into waydroid or how to use weston. I use kde

**Comment by mak448a** [comment]
So if you want to do something like this, I have an idea. @Robbidy since you're on Wayland KDE, you could probably use something like this https://github.com/asweigart/pyautogui/issues/111#issuecomment-1894792488, and hack some code along that moves the mouse back to the start of the screen when you get to the other end. Or find some other way to move the mouse, since this way is not super well documented.

**Comment by Twig6943** [comment]
I've tried using sort of a controller emulator to bypass this issue (I was going to make the mouse act as a right analog stick) but no dice. Still can't play roblox properly because of this issue which really sucks no cap :( (Ik about the workaround using weston but I'm not switching to X just to play roblox) 

(Mentioned issue: https://github.com/waydroid/waydroid/issues/1470)

![image](https://github.com/waydroid/waydroid/assets/119701717/9621937f-88ea-4667-a87d-b38b78428d25)

**Comment by Twig6943** [comment]
does anybody here know how to use the said protocol?
![image](https://github.com/waydroid/waydroid/assets/119701717/8d8aca3c-11d1-4b20-9fb5-fe4ab39b1321)

**Comment by ios7jbpro** [comment]
@Twig6943 waydroid DOES support it. Roblox on mobile doesn't.

**Comment by Twig6943** [comment]
@ios7jbpro Yea yea my bad. Anyways could somebody make an external mouse locker for this use case?

**Comment by ios7jbpro** [comment]
@Twig6943 there already is one, check the comment by no-venv

**Comment by Twig6943** [comment]
That one requires you to run waydroid inside weston tho doesn't it?

**Comment by Beau1437** [comment]
> That one requires you to run waydroid inside weston tho doesn't it?

It does, and it’s not that good because it wraps the cursor around and moves the camera slightly each time. Not to discredit the developer, though, it really is the best you can do as far as an external fix goes.

**Comment by Twig6943** [comment]
So its not possible to use the pointer constraints protocol to make an external mouse locker and so thats why the dev used X stuff (weston,libraries..)?

**Comment by yorunoken** [comment]
just butting in to say I have a project that doesn't lock the cursor, but it keeps the cursor inside of the current monitor you're using: https://github.com/yorunoken/HyprRobloxLock-GUI. There's both a CLI and a GUI version of it,
Only works on Hyprland unfortunately
